src/store/index.ts

```diff
@@ -88,33 +102,70 @@ export default createStore<State>({
       if (res.status !== 200) {
         return "error";
       }
-      await dispatch("decryptUploadResponse", res.data);
+      const msg = await dispatch("decryptResponse", res.data);
+
+      if (!msg) {
+        // TODO: Error Handling
+        return "error";
+      }
+
+      await dispatch("parseUploadMessage", msg);
     },
-    async decryptUploadResponse(
-      { state, dispatch, getters },
-      response: UploadResponse
-    ) {
+    async postAccessForm({ dispatch, commit }, request: UploadRequest) {
+      const res = await axios.post<UploadResponse>(
+        "https://rtc-data.registree.io/auth/tokens",
+        request
+      );
+      if (res.status !== 200) {
+        return "error";
+      }
+      const msg = await dispatch("decryptResponse", res.data);
+
+      if (!msg) {
+        // TODO: Error Handling
+        return "error";
+      }
+
+      //TODO: Open Dialog Box show message
+
+      commit("setDecryptedMsg", msg);
+    },
+    async decryptResponse({ state, getters }, response: UploadResponse) {
       const enclavePubKey = getters.enclavePublicKey;
       if (!enclavePubKey || !state.ourSecretKey) {
         return "error";
       }
-      const msg = decryptMessage(
+      return decryptMessage(
         response.ciphertext,
         response.nonce,
         enclavePubKey,
         state.ourSecretKey
       );
-      if (!msg) {
-        // TODO: Error Handling
-        return "error";
-      }
-      await dispatch("parseUploadMessage", msg);
     },
     async parseUploadMessage({ commit }, message: Uint8Array) {
       console.log("parseUploadMessage:", message);
       const accessKey = btoa(message.slice(0, 24).toString());
       const uuid = message.slice(24);
       commit("saveUploadResult", { accessKey, uuid });
+    },
+    async sealBox({ commit, dispatch, getters }, data) {
```

Comment:
I think it makes more sense to call this action something like `requestExecutionToken`